### PR TITLE
EP-1268: Support new embed web app

### PR DESCRIPTION
**Pull request by MitchellLawson-Burton**
https://getsetup.atlassian.net/browse/EP-1268

Adds experimental support for a new webapp powering the experience.

TODO:
* [x] Update documentation
* [ ] Confirm final URLs for discover / watch pages.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [2.0.0] - TBD
```
Comment (mdbudnick) [inline CHANGELOG.md]: 2.1.0

File: src/index.ts [hunk 2]
```diff
@@ -177,31 +196,44 @@ export function createIframe({
...
   analyticsInfo,
-  classSlug,
+  pageId,
 }: CreateIframeOptions): IframeInstance {
   if (targetPage == 'joinClass' && !sessionId) {
     throw new Error('sessionId is required if you are loading a join class page.')
   }
 
-  if (targetPage == 'class' && !classSlug) {
-    throw new Error('classSlug is required if you are loading a class page.')
```
Comment (mdbudnick) [inline src/index.ts:187]: We are no longer checking for **classSlug** (or **classId**) and it could cause downstream errors now.

**Review by mdbudnick — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 2 above)
Comment (MitchellLawson-Burton) [reply src/index.ts:187]: The `class` page was only a thing for a version of the embed that we don't use any more (at least not with this package).

The `classSlug` was required for the `class` page as the slug was required in the url for that page. But for all the other pages we have the slug only used but the page hosting the iframe, we don't use it inside the iframe anywhere. We just pass it to the hosting page so they can use it for SEO if they want.

But you are correct about checking `classId`, that is required if the page is the `watch` page. Will add check for that.

**Comment by MitchellLawson-Burton** [comment]
URLs for watch / discover pages might still change in the future, but can update then if we need to.

File: README.md [hunk 3]
```diff
@@ -73,13 +71,7 @@ export interface CreateIframeOptions {
   sessionId?: string
 
   /** The id of your organisation as issued to you by GetSetUp. */
-  embeddingOrgId: string
-
-  /**
-   * A stable id for the device the user is using to access the parent page. Used to report analytics back to your organisation.
-   * @deprecated Use {@link analyticsInfo} instead. This will be removed in a future version.
-   */
-  deviceId?: string
```
Comment (MitchellLawson-Burton) [inline README.md:82]: Removed these from the readme because they are deprecated (still in the TSDocs on the interface itself though).

**Review by mdbudnick — APPROVED** [review]
I don't understand what's going on 100%, but with the context these changes look rational to me.

**Pull request merged**